ReportingServicesTools/Functions/CatalogItems/New-RsScheduleXML.ps1

```diff
@@ -0,0 +1,198 @@
+
+        .EXAMPLE
+            New-RsScheduleXML -Weekly -Interval 2 -DaysOfWeek Monday,Tuesday,Wednesday,Thursday,Friday
+
+            Description
+            -----------
+            Creates an XML string to schedule a subscription to run every 2 weeks on weekdays at the current time.
+       
+       .EXAMPLE
+            New-RsScheduleXML -Monthly -DaysOfMonth '1-15,20' -Months January,March,May -Start 5pm -End 12/12/2017
+
+            Description
+            -----------
+            Creates an XML string to schedule a subscription to run on the 1st-15th and 20th of January, March and 
+            May at 5pm until 12th December 2017.
+            
+            Note that -DaysOfMonth is always provided a single string.
+
+        .EXAMPLE
+            New-RsScheduleXML -MonthlyDayOfWeek -DaysOfWeek Saturday -WeekOfMonth LastWeek -Months January,July
+
+            Description
+            -----------
+            Creates an XML string to schedule a subscription to run on the last Saturday of January and July.
+    #>
+
+    [cmdletbinding(SupportsShouldProcess = $true, ConfirmImpact = 'Low',DefaultParameterSetName='Once')]
+    [OutputType(‘System.String’)]
+    Param(
+        [Parameter(ParameterSetName='Minute')]
+        [Switch]      
+        $Minute,
+        
+        [Parameter(ParameterSetName='Daily')]
+        [Switch]      
+        $Daily,
+
+        [Parameter(ParameterSetName='Weekly')]
+        [Switch]      
+        $Weekly,
+
+        [Parameter(ParameterSetName='Monthly')]
+        [Switch]      
+        $Monthly,
+
+        [Parameter(ParameterSetName='MonthlyDOW')]
+        [Switch]      
+        $MonthlyDayOfWeek,
+        
+        [Parameter(ParameterSetName='Once')]
+        [Switch]      
+        $Once,
+
+        [Parameter(ParameterSetName='Minute',Position=0)]
+        [Parameter(ParameterSetName='Daily',Position=0)]
+        [Parameter(ParameterSetName='Weekly',Position=0)]
+        [Int]
+        $Interval = 1,
+
+        [Parameter(ParameterSetName='Weekly')]
+        [Parameter(ParameterSetName='MonthlyDOW')]
+        [ValidateSet('Sunday','Monday','Tuesday','Wednesday','Thursday','Friday','Saturday')]
+        [String[]]
+        $DaysOfWeek,
+
+        [Parameter(ParameterSetName='Monthly')]
+        [Parameter(ParameterSetName='MonthlyDOW')]
+        [ValidateSet('January','February','March','April','May','June','July','August','September','October','November','December')]
+        [String[]]
+        $Months,
+
+        [Parameter(ParameterSetName='Monthly',Mandatory=$True)]
+        [String]
+        $DaysOfMonth,
+
+        [Parameter(ParameterSetName='MonthlyDOW',Mandatory=$True)]
+        [ValidateSet('FirstWeek','SecondWeek','ThirdWeek','FourthWeek','LastWeek')]
+        [String]
+        $WeekOfMonth,
+
+        [DateTime]
+        $Start = (Get-Date),
+
+        [DateTime]
+        $End
+
+    )
+    Process {
```

Comment:
Please format using the guidelines we have specified in Readme. Most of the code is easy to read except for the statements where there are multiple | combined with complex logic.

Reply:
Overhauled in next commit to meet style guide and is now much more readable.